src/formatDate.js

```diff
@@ -8,7 +8,37 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const splittedDate = date.split(fromFormat[3]);
+
+  const yearFormat = fromFormat.includes('YY') ? 'YY' : 'YYYY';
+
+  const year = splittedDate[fromFormat.indexOf(yearFormat)];
+  const month = splittedDate[fromFormat.indexOf('MM')];
+  const day = splittedDate[fromFormat.indexOf('DD')];
+
+  const newYearFormat = toFormat.includes('YY') ? 'YY' : 'YYYY';
```

Comment:
Similar to the previous comment, it's safer to explicitly check for 'YYYY' in `toFormat` to avoid potential issues with substring matches.